crates/bevy_ecs/src/schedule/ambiguity_detection.rs

```diff
@@ -0,0 +1,899 @@
+                        SystemStageSegment::ExclusiveAtStart,
+                        self,
+                        world,
+                    )
+                })
+                .collect();
+
+        let before_commands: Vec<SystemOrderAmbiguity> = find_ambiguities(
+            &self.exclusive_before_commands,
+            &ignored_crates,
+            report_level,
+        )
+        .iter()
+        .map(|(system_a_index, system_b_index, component_ids)| {
+            SystemOrderAmbiguity::from_raw(
+                *system_a_index,
+                *system_b_index,
+                component_ids.to_vec(),
+                SystemStageSegment::ExclusiveBeforeCommands,
+                self,
+                world,
+            )
+        })
+        .collect();
+
+        let at_end: Vec<SystemOrderAmbiguity> =
+            find_ambiguities(&self.exclusive_at_end, &ignored_crates, report_level)
+                .iter()
+                .map(|(system_a_index, system_b_index, component_ids)| {
+                    SystemOrderAmbiguity::from_raw(
+                        *system_a_index,
+                        *system_b_index,
+                        component_ids.to_vec(),
+                        SystemStageSegment::ExclusiveAtEnd,
+                        self,
+                        world,
+                    )
+                })
+                .collect();
+
+        let mut ambiguities = Vec::default();
+        ambiguities.extend(at_start);
+        ambiguities.extend(parallel);
+        ambiguities.extend(before_commands);
+        ambiguities.extend(at_end);
+
+        ambiguities
+    }
+
+    /// Returns the number of system order ambiguities between systems in this stage
+    pub fn n_ambiguities(
+        &mut self,
+        world: &mut World,
+        report_level: ExecutionOrderAmbiguities,
+    ) -> usize {
+        self.ambiguities(world, report_level).len()
+    }
+
+    /// Reports all execution order ambiguities between systems
+    pub fn report_ambiguities(
+        &mut self,
+        world: &mut World,
+        report_level: ExecutionOrderAmbiguities,
+    ) {
+        let ambiguities = self.ambiguities(world, report_level);
+        let unresolved_count = ambiguities.len();
+
+        let mut warning_string = String::new();
+
+        if unresolved_count > 0 {
+            // Grammar
+            if unresolved_count == 1 {
+                warning_string += "One of your stages contains 1 pair of systems with unknown order and conflicting data access. \n\
+				You may want to add `.before()` or `.after()` ordering constraints between some of these systems to prevent bugs.\n";
+            } else {
+                warning_string += &format!("One of your stages contains {unresolved_count} pairs of systems with unknown order and conflicting data access. \n\
+				You may want to add `.before()` or `.after()` ordering constraints between some of these systems to prevent bugs.\n");
+            }
```

Comment:
```suggestion
            let plural = if unresolved_count == 1 { "" } else { "s" };
            warning_string += &format!("One of your stages contains {unresolved_count} pair{plural} of systems with unknown order and conflicting data access. \n\
				You may want to add `.before()` or `.after()` ordering constraints between some of these systems to prevent bugs.\n");
            }
```